### ISSUE TITLE
2.1.0 Features

### ISSUE THREAD

**Issue by sunkibaek** [huiseoul/react-native-fit-image#61]
I would like to setup 2.1.0 roadmap as follow. What do you guys think?

## Technical

1. Refactoring
    - [ ] Check if all implementations are as intended
    - [ ] Check if there are any improvements to be made in terms of performance
1. Add `example` app
    - [ ] include a sample react-native app
    - [ ] it will be useful for development, too
1. [ ] Update dependencies
1. [ ] Support local images
1. [ ] Support arbitary width (or height)
    - as suggestion in #75 
1. [ ] Add support for callbacks to be called even after being used inside the library
    - as suggested in #76

## Doc

1. [ ] List of props
    - Make optional, required props clear
1. [ ] Development flow
1. [ ] Related libraries
    - FastImage

## Features

1. [ ] Add `retriable` state
    - press refresh icon to load the image after failure
1. ~Add 'click to enlarge' feature~
    - This might be out of scope of this project
1. [x] Add to reset image size upon rotation -- or any other parent component resize
1. Export calculation logic to be used outside of react-native-fit-image components

## Others

1. [ ] Make a logo image 😎 

Thanks!

cc @originerd

**Comment by kh71** [comment]
i hope next version it can use onPress or zoom Images

**Comment by codewisdom** [comment]
yes to both onPress and Zoom functionality.  I may start work on both of those features in a fork.

**Comment by kh71** [comment]
Yes good new. Thanks 🙏

**Comment by ddikodroid** [comment]
any update on this?
